packages/slider/src/Slider.ts

```diff
@@ -228,194 +206,65 @@ export class Slider extends Focusable {
         `;
     }
 
-    private renderHandle(): TemplateResult {
+    private renderTrackSegment(start: number, end: number): TemplateResult {
+        if (this.variant === 'ramp') {
+            return html``;
+        }
         return html`
             <div
-                id="handle"
-                style=${this.handleStyle}
-                @manage=${streamingListener(
-                    { type: 'pointerdown', fn: this.handlePointerdown },
-                    { type: 'pointermove', fn: this.handlePointermove },
-                    {
-                        type: ['pointerup', 'pointercancel'],
-                        fn: this.handlePointerup,
-                    }
-                )}
+                class="track"
+                style=${styleMap(this.trackSegmentStyles(start, end))}
                 role="presentation"
-            >
-                <input
-                    type="range"
-                    id="input"
-                    min=${this.min}
-                    max=${this.max}
-                    step=${this.step}
-                    .value=${this.value.toString()}
-                    aria-disabled=${ifDefined(
-                        this.disabled ? 'true' : undefined
-                    )}
-                    aria-valuetext=${this.ariaValueText}
-                    @change=${this.onInputChange}
-                    @focus=${this.onInputFocus}
-                    @blur=${this.onInputBlur}
-                />
-            </div>
+            ></div>
         `;
     }
 
     private renderTrack(): TemplateResult {
+        const segments = this.handleController.trackSegments();
+
+        const trackItems = [
+            { id: 'track0', html: this.renderTrackSegment(...segments[0]) },
+            { id: 'ramp', html: this.renderRamp() },
+            { id: 'ticks', html: this.renderTicks() },
+            { id: 'handles', html: this.handleController.render() },
+            ...segments.slice(1).map(([start, end], index) => ({
+                id: `track${index + 1}`,
+                html: this.renderTrackSegment(start, end),
+            })),
+        ];
+
         return html`
             <div @pointerdown=${this.handleTrackPointerdown}>
                 <div id="controls">
-                    ${this.renderTrackLeft()} ${this.renderRamp()}
-                    ${this.renderTicks()} ${this.renderHandle()}
-                    ${this.renderTrackRight()}
+                    ${repeat(
```

Comment:
Is `repeat` to support them changing order? Or something else?

Reply:
Using `repeat` just allowed me to specifically assign an id to each element rendered to make sure that lit can track them. Previously I was using the `slice` directly in the rendering and you were concerned about that affecting the performance on subsequent renders.